File: twoSum.scala
object Solution {
  // 定义一个twoSum函数，接受一个整数数组和一个目标整数，返回一个整数数组
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    // 遍历数组的索引，从0到数组的长度减1
    for (i <- 0 until nums.size) {
      // 遍历数组的索引，从i+1到数组的长度减1
      for (j <- i + 1 until nums.size) {
        // 如果数组中第i个元素和第j个元素的和等于目标整数
        if (nums(i) + nums(j) == target) {
          // 返回一个包含i和j的数组
           
           return Array(i, j)
        }
      }
    }
    // 返回一个空数组
    Array()
  }
}

File: twoSum.rs
// `Point` 是 `(u8, u8)` 的新类型别名
//type Point = (u8, u8);
//let p: Point = (41, 68);
type Solution = two_sum;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        for i in 0..nums.len() { // 枚举 i
            for j in i + 1..nums.len() { // 枚举 i 右边的 j
                if nums[i] + nums[j] == target { // 满足要求
                    return vec![i as i32, j as i32]; // 返回两个数的下标
                }
            }
        }
        unreachable!() // 题目保证有解，循环中一定会 return
    }
}
